File: experimental/scratch-cli-2/src/commands/connections.rs
use std::collections::HashMap;

use anyhow::{bail, Result};
use clap::{Args, Subcommand};
use dialoguer::{Input, Password, Select};

use crate::cli::Cli;
use crate::commands::helpers::*;

#[derive(Args)]
pub struct ConnectionsCommand {
    #[command(subcommand)]
    pub action: ConnectionsAction,

    /// Override workbook auto-detection
    #[arg(long)]
    pub workbook: Option<String>,
}

#[derive(Subcommand)]
pub enum ConnectionsAction {
    /// List all connections in the workbook
    List {
        #[arg(long)]
        json: bool,
    },
    /// Authorize a new connection
    Add {
        #[arg(long)]
        service: Option<String>,
        #[arg(long)]
        name: Option<String>,
        #[arg(long)]
        param: Vec<String>,
        #[arg(long)]
        json: bool,
    },
    /// Show connection details
    Show {
        id: String,
        #[arg(long)]
        json: bool,
    },
    /// Delete a connection
    Remove {
        id: String,
        #[arg(long)]
        yes: bool,
        #[arg(long)]
        json: bool,
    },
}

struct AuthField {
    key: &'static str,
    display_name: &'static str,
    description: &'static str,
    sensitive: bool,
}

fn service_auth_config() -> Vec<(&'static str, Vec<AuthField>)> {
    vec![
        ("AIRTABLE", vec![
            AuthField { key: "apiKey", display_name: "API Key", description: "Your Airtable personal access token", sensitive: true },
        ]),
        ("WEBFLOW", vec![
            AuthField { key: "apiKey", display_name: "API Key", description: "Your Webflow API token", sensitive: true },
        ]),
        ("AUDIENCEFUL", vec![
            AuthField { key: "apiKey", display_name: "API Key", description: "Your Audienceful API key", sensitive: true },
        ]),
        ("SHOPIFY", vec![
            AuthField { key: "shopDomain", display_name: "Shop Domain", description: "Your Shopify store (e.g. my-store or my-store.myshopify.com)", sensitive: false },
            AuthField { key: "apiKey", display_name: "API Key", description: "Your Shopify Admin API access token", sensitive: true },
        ]),
        ("MOCO", vec![
            AuthField { key: "domain", display_name: "Moco Domain", description: "Your Moco subdomain (e.g. yourcompany for yourcompany.mocoapp.com)", sensitive: false },
            AuthField { key: "apiKey", display_name: "API Key", description: "Your Moco API key", sensitive: true },
        ]),
        ("WORDPRESS", vec![
            AuthField { key: "endpoint", display_name: "WordPress URL", description: "The URL of your WordPress site (e.g. https://example.com)", sensitive: false },
            AuthField { key: "username", display_name: "Username", description: "Your WordPress account username", sensitive: false },
            AuthField { key: "password", display_name: "Application Password", description: "Your WordPress application password (generate in Users > Profile)", sensitive: true },
        ]),
        ("POSTGRES", vec![
            AuthField { key: "connectionString", display_name: "Connection String", description: "PostgreSQL connection string (e.g. postgresql://user:pass@host:5432/dbname)", sensitive: true },
        ]),
        ("SUPABASE", vec![
            AuthField { key: "connectionString", display_name: "Connection String", description: "Supabase connection string (find in Settings > Database > Connection string)", sensitive: true },
        ]),
    ]
}

fn supported_services() -> Vec<&'static str> {
    vec!["AIRTABLE", "WEBFLOW", "SHOPIFY", "MOCO", "AUDIENCEFUL", "WORDPRESS", "POSTGRES", "SUPABASE"]
}

fn parse_params(raw: &[String]) -> Result<HashMap<String, String>> {
    let mut params = HashMap::new();
    for p in raw {
        let parts: Vec<&str> = p.splitn(2, '=').collect();
        if parts.len() != 2 {
            bail!("invalid --param format {:?}, expected key=value", p);
        }
        params.insert(parts[0].to_string(), parts[1].to_string());
    }
    Ok(params)
}

impl ConnectionsCommand {
    pub fn run(&self, cli: &Cli) -> Result<()> {
        match &self.action {
            ConnectionsAction::List { json } => run_list(cli, self.workbook.as_deref(), *json),
            ConnectionsAction::Add {
                service,
                name,
                param,
                json,
            } => run_add(cli, self.workbook.as_deref(), service.as_deref(), name.as_deref(), param, *json),
            ConnectionsAction::Show { id, json } => run_show(cli, self.workbook.as_deref(), id, *json),
            ConnectionsAction::Remove { id, yes, json } => run_remove(cli, self.workbook.as_deref(), id, *yes, *json),
        }
    }
}

fn run_list(cli: &Cli, workbook_flag: Option<&str>, json_output: bool) -> Result<()> {
    let client = get_authenticated_client(cli)?;
    let workbook_id = resolve_workbook_context(workbook_flag)?;
    let connections = client.list_connections(&workbook_id)?;

    if json_output {
        return print_json(&connections);
    }

    if connections.is_empty() {
        println!("No connections found in this workbook.");
        println!();
        println!("Add a connection with: scratchmd connections add");
        return Ok(());
    }

    println!();
    println!(
        "  {:<36}  {:<14}  {:<20}  {:<8}  {}",
        "ID", "SERVICE", "NAME", "HEALTH", "CREATED"
    );
    println!(
        "  {:<36}  {:<14}  {:<20}  {:<8}  {}",
        "----", "-------", "----", "------", "-------"
    );
    for c in &connections {
        let health = c.health_status.as_deref().unwrap_or("-");
        let created = if c.created_at.len() > 10 {
            &c.created_at[..10]
        } else {
            &c.created_at
        };
        let name = if c.display_name.len() > 20 {
            format!("{}...", &c.display_name[..17])
        } else {
            c.display_name.clone()
        };
        println!(
            "  {:<36}  {:<14}  {:<20}  {:<8}  {}",
            c.id, c.service, name, health, created
        );
    }
    println!();

    Ok(())
}

fn run_add(
    cli: &Cli,
    workbook_flag: Option<&str>,
    service: Option<&str>,
    name: Option<&str>,
    raw_params: &[String],
    json_output: bool,
) -> Result<()> {
    let client = get_authenticated_client(cli)?;
    let workbook_id = resolve_workbook_context(workbook_flag)?;

    if let (Some(svc), true) = (service, !raw_params.is_empty()) {
        // Non-interactive mode
        let svc = svc.to_uppercase();
        let params = parse_params(raw_params)?;

        let configs = service_auth_config();
        if let Some((_, fields)) = configs.iter().find(|(s, _)| *s == svc) {
            let missing: Vec<_> = fields
                .iter()
                .filter(|f| !params.contains_key(f.key))
                .map(|f| format!("{} ({})", f.key, f.display_name))
                .collect();
            if !missing.is_empty() {
                bail!("missing required params for {}: {}", svc, missing.join(", "));
            }
        }

        let result = client.create_connection(
            &workbook_id,
            &svc,
            name,
            params,
        )?;

        if json_output {
            return print_json(&result);
        }
        println!("\nConnection '{}' created successfully.", result.display_name);
        println!("  ID:      {}", result.id);
        println!("  Service: {}", result.service);
        if let Some(ref h) = result.health_status {
            println!("  Health:  {}", h);
        }
        println!();
        return Ok(());
    }

    // Interactive mode
    let services = supported_services();
    let selection = Select::new()
        .with_prompt("Select a service")
        .items(&services)
        .interact()?;
    let selected_service = services[selection];

    let configs = service_auth_config();
    let (_, fields) = configs
        .iter()
        .find(|(s, _)| *s == selected_service)
        .ok_or_else(|| anyhow::anyhow!("unknown service {:?}", selected_service))?;

    let mut params = HashMap::new();
    for field in fields {
        let value = if field.sensitive {
            Password::new()
                .with_prompt(format!("{} ({})", field.display_name, field.description))
                .interact()?
        } else {
            Input::<String>::new()
                .with_prompt(format!("{} ({})", field.display_name, field.description))
                .interact_text()?
        };
        let value = value.trim().to_string();
        if value.is_empty() {
            bail!("{} cannot be empty", field.display_name);
        }
        params.insert(field.key.to_string(), value);
    }

    let lower = selected_service.to_lowercase();
    let default_name = format!("{}{}", &lower[..1].to_uppercase(), &lower[1..]);
    let display_name: String = Input::new()
        .with_prompt("Display name")
        .default(default_name.clone())
        .interact_text()?;
    let display_name = if display_name.is_empty() {
        default_name
    } else {
        display_name
    };

    let result = client.create_connection(
        &workbook_id,
        selected_service,
        Some(&display_name),
        params,
    )?;

    if json_output {
        return print_json(&result);
    }
    println!("\nConnection '{}' created successfully.", result.display_name);
    println!("  ID:      {}", result.id);
    println!("  Service: {}", result.service);
    if let Some(ref h) = result.health_status {
        println!("  Health:  {}", h);
    }
    println!();
    Ok(())
}

fn run_show(cli: &Cli, workbook_flag: Option<&str>, id: &str, json_output: bool) -> Result<()> {
    let client = get_authenticated_client(cli)?;
    let workbook_id = resolve_workbook_context(workbook_flag)?;
    let connection = client.get_connection(&workbook_id, id)?;

    if json_output {
        return print_json(&connection);
    }

    let health = connection.health_status.as_deref().unwrap_or("-");

    println!();
    println!("  Name:    {}", connection.display_name);
    println!("  ID:      {}", connection.id);
    println!("  Service: {}", connection.service);
    println!("  Auth:    {}", connection.auth_type);
    println!("  Health:  {}", health);
    if let Some(ref msg) = connection.health_status_message {
        if !msg.is_empty() {
            println!("  Message: {}", msg);
        }
    }
    println!("  Created: {}", connection.created_at);
    println!("  Updated: {}", connection.updated_at);
    println!();

    Ok(())
}

fn run_remove(cli: &Cli, workbook_flag: Option<&str>, id: &str, yes: bool, json_output: bool) -> Result<()> {
    let client = get_authenticated_client(cli)?;
    let workbook_id = resolve_workbook_context(workbook_flag)?;
    let connection = client.get_connection(&workbook_id, id)?;

    if !yes && !json_output {
        if !confirm(&format!(
            "Are you sure you want to delete connection \"{}\" ({})? [y/N] ",
            connection.display_name, connection.id
        ))? {
            println!("Cancelled.");
            return Ok(());
        }
    }

    client.delete_connection(&workbook_id, id)?;

    if json_output {
        return print_json(&serde_json::json!({
            "success": true,
            "id": connection.id,
            "displayName": connection.display_name,
        }));
    }

    println!(
        "Connection \"{}\" deleted successfully.",
        connection.display_name
    );
    Ok(())
}

File: scratch-cli/internal/cmd/connections.go
package cmd

import (
	"bufio"
	"encoding/json"
	"fmt"
	"os"
	"strings"

	survey "github.com/AlecAivazis/survey/v2"
	"github.com/spf13/cobra"
	"github.com/whalesync/scratch-cli/internal/api"
)

// authField describes a credential field required by a service.
type authField struct {
	Key         string // Key sent in userProvidedParams (e.g. "apiKey", "shopDomain")
	DisplayName string // Human-readable label for prompts
	Description string // Help text shown in interactive mode
	Required    bool
	Sensitive   bool // If true, input is masked (password prompt)
}

// serviceAuthConfig maps each supported service to its required credential fields.
var serviceAuthConfig = map[string][]authField{
	"WEBFLOW": {
		{Key: "apiKey", DisplayName: "API Key", Description: "Your Webflow API token", Required: true, Sensitive: true},
	},
	"AUDIENCEFUL": {
		{Key: "apiKey", DisplayName: "API Key", Description: "Your Audienceful API key", Required: true, Sensitive: true},
	},
	"SHOPIFY": {
		{Key: "shopDomain", DisplayName: "Shop Domain", Description: "Your Shopify store (e.g. my-store or my-store.myshopify.com)", Required: true, Sensitive: false},
		{Key: "apiKey", DisplayName: "API Key", Description: "Your Shopify Admin API access token", Required: true, Sensitive: true},
	},
	"MOCO": {
		{Key: "domain", DisplayName: "Moco Domain", Description: "Your Moco subdomain (e.g. yourcompany for yourcompany.mocoapp.com)", Required: true, Sensitive: false},
		{Key: "apiKey", DisplayName: "API Key", Description: "Your Moco API key", Required: true, Sensitive: true},
	},
	"WORDPRESS": {
		{Key: "endpoint", DisplayName: "WordPress URL", Description: "The URL of your WordPress site (e.g. https://example.com)", Required: true, Sensitive: false},
		{Key: "username", DisplayName: "Username", Description: "Your WordPress account username", Required: true, Sensitive: false},
		{Key: "password", DisplayName: "Application Password", Description: "Your WordPress application password (generate in Users > Profile)", Required: true, Sensitive: true},
	},
	"AIRTABLE": {
		{Key: "apiKey", DisplayName: "API Key", Description: "Your Airtable personal access token", Required: true, Sensitive: true},
	},
	"POSTGRES": {
		{Key: "connectionString", DisplayName: "Connection String", Description: "PostgreSQL connection string (e.g. postgresql://user:pass@host:5432/dbname)", Required: true, Sensitive: true},
	},
	"SUPABASE": {
		{Key: "connectionString", DisplayName: "Connection String", Description: "Supabase connection string (find in Settings > Database > Connection string)", Required: true, Sensitive: true},
	},
	"PIPEDRIVE": {
		{Key: "apiKey", DisplayName: "API Key", Description: "Your Pipedrive API token (find in Settings > Personal preferences > API)", Required: true, Sensitive: true},
	},
}

// supportedServices returns the list of service names in display order.
func supportedServices() []string {
	return []string{"AIRTABLE", "WEBFLOW", "SHOPIFY", "MOCO", "AUDIENCEFUL", "WORDPRESS", "POSTGRES", "SUPABASE", "PIPEDRIVE"}
}

var connectionsCmd = &cobra.Command{
	Use:   "connections",
	Short: "Manage connections",
	Long: `Manage connections (connector accounts) in a workspace.

Commands:
  connections list      List all connections
  connections add       Authorize a new connection
  connections show      Show connection details
  connections remove    Delete a connection`,
}

var connectionsListCmd = &cobra.Command{
	Use:   "list",
	Short: "List all connections in the workspace",
	Long: `List all connections in a workspace.

If run inside a workspace directory (contains .scratchmd marker), the workspace
is detected automatically. Otherwise, use the --workspace flag.

Examples:
  scratchmd connections list
  scratchmd connections list --workspace wb_abc123`,
	RunE: runConnectionsList,
}

var connectionsAddCmd = &cobra.Command{
	Use:   "add",
	Short: "Authorize a new connection",
	Long: `Create a new connection in the workspace.

In interactive mode (no flags), you will be prompted for the service and
its required credentials. In non-interactive mode, provide --service and
--param flags for each required credential field.

Non-interactive examples:
  scratchmd connections add --service AIRTABLE --param apiKey=<token>
  scratchmd connections add --service WEBFLOW --param apiKey=<key>
  scratchmd connections add --service SHOPIFY --param shopDomain=my-store --param apiKey=<key>
  scratchmd connections add --service WORDPRESS --param endpoint=https://example.com --param username=admin --param password=<app-password>
  scratchmd connections add --service MOCO --param domain=yourcompany --param apiKey=<key>
  scratchmd connections add --service PIPEDRIVE --param apiKey=<token>
  scratchmd connections add --service POSTGRES --param connectionString=postgresql://user:pass@host:5432/db
  scratchmd connections add --service SUPABASE --param connectionString=postgresql://postgres.[ref]:[password]@aws-1-us-east-1.pooler.supabase.com:6543/postgres`,
	RunE: runConnectionsAdd,
}

var connectionsShowCmd = &cobra.Command{
	Use:   "show <id>",
	Short: "Show connection details",
	Long: `Show details for a connection.

Examples:
  scratchmd connections show conn_abc123
  scratchmd connections show conn_abc123 --workspace wb_abc123`,
	Args: cobra.ExactArgs(1),
	RunE: runConnectionsShow,
}

var connectionsRemoveCmd = &cobra.Command{
	Use:   "remove <id>",
	Short: "Delete a connection",
	Long: `Delete a connection from the workspace.

Examples:
  scratchmd connections remove conn_abc123
  scratchmd connections remove conn_abc123 --yes`,
	Args: cobra.ExactArgs(1),
	RunE: runConnectionsRemove,
}

func init() {
	rootCmd.AddCommand(connectionsCmd)
	connectionsCmd.PersistentFlags().String("workspace", "", "Workspace ID (auto-detected from .scratchmd if not set)")
	connectionsCmd.PersistentFlags().String("workbook", "", "Workspace ID (auto-detected from .scratchmd if not set)")
	connectionsCmd.PersistentFlags().MarkHidden("workbook")

	connectionsCmd.AddCommand(connectionsListCmd)
	connectionsCmd.AddCommand(connectionsAddCmd)
	connectionsCmd.AddCommand(connectionsShowCmd)
	connectionsCmd.AddCommand(connectionsRemoveCmd)

	// --json flag on each subcommand
	connectionsListCmd.Flags().Bool("json", false, "Output as JSON")
	connectionsAddCmd.Flags().Bool("json", false, "Output as JSON")
	connectionsShowCmd.Flags().Bool("json", false, "Output as JSON")
	connectionsRemoveCmd.Flags().Bool("json", false, "Output as JSON")

	// Command-specific flags
	connectionsRemoveCmd.Flags().Bool("yes", false, "Skip confirmation prompt")
	connectionsAddCmd.Flags().String("service", "", "Service type (AIRTABLE, WEBFLOW, SHOPIFY, MOCO, AUDIENCEFUL, WORDPRESS, POSTGRES, SUPABASE, PIPEDRIVE)")
	connectionsAddCmd.Flags().StringSlice("param", nil, "Credential parameter as key=value (repeatable)")
	connectionsAddCmd.Flags().String("name", "", "Display name for the connection")
}

func runConnectionsList(cmd *cobra.Command, args []string) error {
	jsonOutput, _ := cmd.Flags().GetBool("json")

	client, err := getAuthenticatedClient()
	if err != nil {
		return err
	}

	workbookID, err := resolveWorkbookContext(cmd)
	if err != nil {
		return err
	}

	connections, err := client.ListConnections(workbookID)
	if err != nil {
		return fmt.Errorf("failed to list connections: %w", err)
	}

	if jsonOutput {
		encoder := json.NewEncoder(os.Stdout)
		encoder.SetIndent("", "  ")
		return encoder.Encode(connections)
	}

	if len(connections) == 0 {
		fmt.Println("No connections found in this workspace.")
		fmt.Println()
		fmt.Println("Add a connection with: scratchmd connections add")
		return nil
	}

	fmt.Println()
	fmt.Printf("  %-36s  %-14s  %-20s  %-8s  %s\n", "ID", "SERVICE", "NAME", "HEALTH", "CREATED")
	fmt.Printf("  %-36s  %-14s  %-20s  %-8s  %s\n", "----", "-------", "----", "------", "-------")
	for _, c := range connections {
		health := "-"
		if c.HealthStatus != nil {
			health = *c.HealthStatus
		}
		created := c.CreatedAt
		if len(created) > 10 {
			created = created[:10]
		}
		name := c.DisplayName
		if len(name) > 20 {
			name = name[:17] + "..."
		}
		fmt.Printf("  %-36s  %-14s  %-20s  %-8s  %s\n", c.ID, c.Service, name, health, created)
	}
	fmt.Println()

	return nil
}

// parseParams parses --param key=value flags into a map.
func parseParams(raw []string) (map[string]string, error) {
	params := make(map[string]string)
	for _, p := range raw {
		parts := strings.SplitN(p, "=", 2)
		if len(parts) != 2 {
			return nil, fmt.Errorf("invalid --param format %q, expected key=value", p)
		}
		params[parts[0]] = parts[1]
	}
	return params, nil
}

func runConnectionsAdd(cmd *cobra.Command, args []string) error {
	jsonOutput, _ := cmd.Flags().GetBool("json")

	client, err := getAuthenticatedClient()
	if err != nil {
		return err
	}

	workbookID, err := resolveWorkbookContext(cmd)
	if err != nil {
		return err
	}

	service, _ := cmd.Flags().GetString("service")
	rawParams, _ := cmd.Flags().GetStringSlice("param")
	name, _ := cmd.Flags().GetString("name")

	if service != "" && len(rawParams) > 0 {
		// Non-interactive mode
		service = strings.ToUpper(service)
		params, err := parseParams(rawParams)
		if err != nil {
			return err
		}

		// Validate required fields if we know this service
		if fields, ok := serviceAuthConfig[service]; ok {
			var missing []string
			for _, f := range fields {
				if f.Required {
					if _, exists := params[f.Key]; !exists {
						missing = append(missing, fmt.Sprintf("%s (%s)", f.Key, f.DisplayName))
					}
				}
			}
			if len(missing) > 0 {
				return fmt.Errorf("missing required params for %s: %s", service, strings.Join(missing, ", "))
			}
		}

		req := &api.CreateConnectionRequest{
			Service:            service,
			DisplayName:        name,
			UserProvidedParams: params,
		}

		result, err := client.CreateConnection(workbookID, req)
		if err != nil {
			return fmt.Errorf("failed to create connection: %w", err)
		}

		return printConnectionCreated(result, jsonOutput)
	}

	// Interactive mode
	var selectedService string
	servicePrompt := &survey.Select{
		Message: "Select a service:",
		Options: supportedServices(),
	}
	if err := survey.AskOne(servicePrompt, &selectedService); err != nil {
		return fmt.Errorf("prompt cancelled: %w", err)
	}

	fields, ok := serviceAuthConfig[selectedService]
	if !ok {
		return fmt.Errorf("unknown service %q", selectedService)
	}

	// Collect credential fields
	params := make(map[string]string)
	for _, field := range fields {
		var value string
		if field.Sensitive {
			prompt := &survey.Password{
				Message: fmt.Sprintf("%s:", field.DisplayName),
				Help:    field.Description,
			}
			if err := survey.AskOne(prompt, &value); err != nil {
				return fmt.Errorf("prompt cancelled: %w", err)
			}
		} else {
			prompt := &survey.Input{
				Message: fmt.Sprintf("%s:", field.DisplayName),
				Help:    field.Description,
			}
			if err := survey.AskOne(prompt, &value); err != nil {
				return fmt.Errorf("prompt cancelled: %w", err)
			}
		}

		value = strings.TrimSpace(value)
		if field.Required && value == "" {
			return fmt.Errorf("%s cannot be empty", field.DisplayName)
		}
		if value != "" {
			params[field.Key] = value
		}
	}

	// Prompt for display name
	lower := strings.ToLower(selectedService)
	defaultName := strings.ToUpper(lower[:1]) + lower[1:]
	var displayName string
	namePrompt := &survey.Input{
		Message: "Display name:",
		Default: defaultName,
	}
	if err := survey.AskOne(namePrompt, &displayName); err != nil {
		return fmt.Errorf("prompt cancelled: %w", err)
	}
	if displayName == "" {
		displayName = defaultName
	}

	req := &api.CreateConnectionRequest{
		Service:            selectedService,
		DisplayName:        displayName,
		UserProvidedParams: params,
	}

	result, err := client.CreateConnection(workbookID, req)
	if err != nil {
		return fmt.Errorf("failed to create connection: %w", err)
	}

	return printConnectionCreated(result, jsonOutput)
}

func printConnectionCreated(result *api.Connection, jsonOutput bool) error {
	if jsonOutput {
		encoder := json.NewEncoder(os.Stdout)
		encoder.SetIndent("", "  ")
		return encoder.Encode(result)
	}

	fmt.Printf("\nConnection '%s' created successfully.\n", result.DisplayName)
	fmt.Printf("  ID:      %s\n", result.ID)
	fmt.Printf("  Service: %s\n", result.Service)
	if result.HealthStatus != nil {
		fmt.Printf("  Health:  %s\n", *result.HealthStatus)
	}
	if result.HealthStatusMessage != nil && *result.HealthStatusMessage != "" {
		fmt.Printf("  Message: %s\n", *result.HealthStatusMessage)
	}
	fmt.Println()
	return nil
}

func runConnectionsShow(cmd *cobra.Command, args []string) error {
	jsonOutput, _ := cmd.Flags().GetBool("json")

	client, err := getAuthenticatedClient()
	if err != nil {
		return err
	}

	workbookID, err := resolveWorkbookContext(cmd)
	if err != nil {
		return err
	}

	connectionID := args[0]

	connection, err := client.GetConnection(workbookID, connectionID)
	if err != nil {
		return fmt.Errorf("failed to get connection: %w", err)
	}

	if jsonOutput {
		encoder := json.NewEncoder(os.Stdout)
		encoder.SetIndent("", "  ")
		return encoder.Encode(connection)
	}

	fmt.Println()
	fmt.Printf("  Name:    %s\n", connection.DisplayName)
	fmt.Printf("  ID:      %s\n", connection.ID)
	fmt.Printf("  Service: %s\n", connection.Service)
	fmt.Printf("  Auth:    %s\n", connection.AuthType)
	health := "-"
	if connection.HealthStatus != nil {
		health = *connection.HealthStatus
	}
	fmt.Printf("  Health:  %s\n", health)
	if connection.HealthStatusMessage != nil && *connection.HealthStatusMessage != "" {
		fmt.Printf("  Message: %s\n", *connection.HealthStatusMessage)
	}
	fmt.Printf("  Created: %s\n", connection.CreatedAt)
	fmt.Printf("  Updated: %s\n", connection.UpdatedAt)
	fmt.Println()

	return nil
}

func runConnectionsRemove(cmd *cobra.Command, args []string) error {
	jsonOutput, _ := cmd.Flags().GetBool("json")
	yes, _ := cmd.Flags().GetBool("yes")

	client, err := getAuthenticatedClient()
	if err != nil {
		return err
	}

	workbookID, err := resolveWorkbookContext(cmd)
	if err != nil {
		return err
	}

	connectionID := args[0]

	// Get details for confirmation
	connection, err := client.GetConnection(workbookID, connectionID)
	if err != nil {
		return fmt.Errorf("failed to get connection: %w", err)
	}

	// Confirmation prompt
	if !yes && !jsonOutput {
		fmt.Printf("Are you sure you want to delete connection \"%s\" (%s)? [y/N] ", connection.DisplayName, connection.ID)
		reader := bufio.NewReader(os.Stdin)
		response, err := reader.ReadString('\n')
		if err != nil {
			return fmt.Errorf("failed to read response: %w", err)
		}
		response = strings.TrimSpace(strings.ToLower(response))
		if response != "y" && response != "yes" {
			fmt.Println("Cancelled.")
			return nil
		}
	}

	if err := client.DeleteConnection(workbookID, connectionID); err != nil {
		return fmt.Errorf("failed to delete connection: %w", err)
	}

	if jsonOutput {
		encoder := json.NewEncoder(os.Stdout)
		encoder.SetIndent("", "  ")
		return encoder.Encode(map[string]interface{}{
			"success": true,
			"id":      connectionID,
			"name":    connection.DisplayName,
		})
	}

	fmt.Printf("Connection \"%s\" deleted successfully.\n", connection.DisplayName)

	return nil
}
